### ISSUE TITLE
sway does not start with EGL_NOT_INITIALIZED error

### ISSUE THREAD

**Issue by lunik1** [nix-community/home-manager#2150]
### Issue description

When trying to launch sway (from tty) I get the error

```
libEGL warning: MESA-LOADER: failed to open : /run/opengl-driver/lib/dri/_dri.so: cannot open shared object file: No such file or directory (search paths /run/opengl-driver/lib/dri)

00:00:00.276 [ERROR] [wlr] [EGL] command: eglInitialize, error: EGL_NOT_INITIALIZED (0x3001), message: "DRI2: failed to load driver"
libEGL warning: MESA-LOADER: failed to open : /run/opengl-driver/lib/dri/_dri.so: cannot open shared object file: No such file or directory (search paths /run/opengl-driver/lib/dri)

00:00:00.276 [ERROR] [wlr] [EGL] command: eglInitialize, error: EGL_NOT_INITIALIZED (0x3001), message: "DRI2: failed to load driver"
00:00:00.276 [ERROR] [wlr] [EGL] command: eglInitialize, error: EGL_NOT_INITIALIZED (0x3001), message: "eglInitialize"
00:00:00.276 [ERROR] [wlr] [render/egl.c:217] Failed to initialize EGL
00:00:00.276 [ERROR] [wlr] [EGL] command: eglMakeCurrent, error: EGL_BAD_DISPLAY (0x3008), message: "Invalid display (nil)"
00:00:00.276 [ERROR] [wlr] [render/wlr_renderer.c:263] Could not initialize EGL
00:00:00.283 [ERROR] [wlr] [backend/drm/renderer.c:34] Failed to create EGL/WLR renderer
00:00:00.291 [ERROR] [wlr] [backend/drm/backend.c:201] Failed to initialize renderer
00:00:00.293 [ERROR] [wlr] [backend/backend.c:174] Failed to create DRM backend
00:00:00.293 [ERROR] [wlr] [backend/backend.c:312] Failed to open any DRM device
00:00:00.371 [ERROR] [sway/server.c:55] Unable to create backend
```

(It seems strange that sway is trying to use `_dri.so`)

I am using `home-manager` to manage sway, tracking nixpkgs-unstable but my system configuration tracks 21.05. My flake-based config can be found [here](https://gitlab.com/lunik1/nix-config). My system has intel graphics.

I'm not completely sure whether this is nixpkgs or home-manager's issue, but it seems to have occurred when bumping nixpkgs from NixOS/nixpkgs@51bb9f3e9ab6161a3bf7746e20b955712cef618b to NixOS/nixpkgs@b27eaa18b47b77fbbf9f3bd7a17046b2d64b637b. If I pin nixpkgs at NixOS/nixpkgs@b27eaa18b47b77fbbf9f3bd7a17046b2d64b637b and just update home-manager sway still runs. I am working on finding the specific commit with the breakage.

#### Maintainer CC

@alexarice @sumnerevans

#### Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 5.12.9-zen1, NixOS, 21.05.20210623.0ccd0d9 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `relaxed`
 - version: `nix-env (Nix) 2.4pre20210601_5985b8b`
 - channels(root): `"nixos-20.09.3332.d303eee16ce, nixos-unstable-21.05pre272380.f6b5bfdb470"`
 - channels(corin): `"nixpkgs-21.05pre286925.3e62e383f4b"`
 - nixpkgs: `/home/corin/.nix-defexpr/channels/nixpkgs`

**Comment by berbiche** [comment]
A starting point to help you in your research: https://github.com/NixOS/nixpkgs/pull/128459

**Comment by Misterio77** [comment]
I think i got the same error fixed by setting `hardware.opengl.enable = true` on `configuration.nix`.

**Comment by berbiche** [comment]
@lunik1 did @Misterio77's solution work?

**Comment by lunik1** [comment]
No. There is some more information on  NixOS/nixpkgs#128469. It is caused by running the home-manger nixpkgs and system nixpkgs on different versions. For me setting the sway package to null in home-manager and using the sway from the NixOS module worked.
